akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/PersistentBehaviorImpl.scala

```diff
@@ -76,21 +79,23 @@ private[akka] final case class PersistentBehaviorImpl[Command, Event, State](
           recovery,
           holdingRecoveryPermit = false,
           settings = settings,
-          internalStash = internalStash
+          stashState = stashState
         )
 
         // needs to accept Any since we also can get messages from the journal
         // not part of the protocol
         val onStopInterceptor = new BehaviorInterceptor[Any, Any] {
+
           import BehaviorInterceptor._
+
           def aroundReceive(ctx: typed.ActorContext[Any], msg: Any, target: ReceiveTarget[Any]): Behavior[Any] = {
             target(ctx, msg)
           }
 
           def aroundSignal(ctx: typed.ActorContext[Any], signal: Signal, target: SignalTarget[Any]): Behavior[Any] = {
             if (signal == PostStop) {
               eventsourcedSetup.cancelRecoveryTimer()
-              clearStashBuffer()
+              stashState.clearStashBuffers()
```

Comment:
So we went with clearing both stashes and users have to put it in their state if they want to preserve a buffer in their domain?

Reply:
Ah but this is just PostStop, not PreRestart